sync-todo/v2/client/cpp/managers/auth_manager.cpp

```diff
@@ -0,0 +1,21 @@
+#include "auth_manager.hpp"
+
+AuthManager::AuthManager(Delegate *delegate, realm::App *app): _delegate(delegate), _app(app) {  }
+
+void AuthManager::registerUser(std::string const& userEmail, std::string const& userPassword) {
+  _app->register_user(userEmail, userPassword).get();
+  logIn(userEmail, userPassword);
+}
+
+void AuthManager::logIn(std::string const& userEmail, std::string const& userPassword) {
+  _app->login(
+          realm::App::credentials::username_password(userEmail, userPassword))
+      .get();
+  _delegate->onLoggedIn();
+}
+
+void AuthManager::logOut() {
+  auto currentUser = _app->get_current_user();
+  currentUser->log_out().get();
```

Comment:
This function is used in a `Logout` button call in the Home Controller, which isn't displayed unless there is a `currentUser`. Otherwise, the user only sees the Login screen. So theoretically it shouldn't be possible for this to be null. Should I add a null check anyway?

Reply:
Yes because there's nothing stopping someone from calling this anyway, and it may be harder to debug a crash from that (when currentUser might be garbage rather than null) than an assert(). (Btw, I would just [assert()](https://en.cppreference.com/w/cpp/error/assert))